### PR TITLE
[refactor] refactored test setup and added more common exceptions

### PR DESCRIPTION
**Pull request by leeuwis**

### CODE REVIEW

File: test/e2e.test.ts [hunk 1]
```diff
@@ -124,6 +142,19 @@ describe('E2E', () => {
         })
       ));
 
+  it('read', () =>
```
Comment (ggurkal) [inline test/e2e.test.ts:145]: The test name should be updated here.

File: lib/exceptions/BadRequestException.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { HttpException } from './HttpException';
 export class BadRequestException extends HttpException {
   public name = 'BadRequestException';
 
-  public constructor(message?: string, errors?: string[]) {
+  public constructor(message: string = 'Bad request', errors?: string[]) {
```
Comment (ggurkal) [inline lib/exceptions/BadRequestException.ts:6]: I believe default messages should reflect the web standards, thus `Bad Request` would be the correct message here.

File: lib/exceptions/InternalServerErrorException.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { HttpException } from './HttpException';
+
+export class InternalServerErrorException extends HttpException {
+  public name = 'InternalServerErrorException';
+
+  public constructor(message: string = 'Internal server error') {
```
Comment (ggurkal) [inline lib/exceptions/InternalServerErrorException.ts:6]: Should be `Internal Server Error`.

File: lib/exceptions/NotFoundException.ts [hunk 4]
```diff
@@ -1,9 +1,9 @@
 import { HttpException } from './HttpException';
 
 export class NotFoundException extends HttpException {
-  public name = 'BadRequestException';
+  public name = 'NotFoundException';
 
-  public constructor(message?: string) {
+  public constructor(message: string = 'Not found') {
```
Comment (ggurkal) [inline lib/exceptions/NotFoundException.ts:6]: Should be `Not Found`.

**Comment by ggurkal** [comment]
From my point of view, it's better to cover all the possible scenarios in E2E tests, but it's not manageable to do that in one test file. Therefore, multiple E2E tests are needed and I think it's cleaner and more organized to have a dedicated `test` folder for all those things.

I'm neutral about unit tests though... But, if we're going to have a separate folder for E2E tests, why not moving all tests to that folder 🤔

**Comment by mitchelvanbever** [comment]
> From my point of view, it's better to cover all the possible scenarios in E2E tests, but it's not manageable to do that in one test file. Therefore, multiple E2E tests are needed and I think it's cleaner and more organized to have a dedicated `test` folder for all those things.
> 
> I'm neutral about unit tests though... But, if we're going to have a separate folder for E2E tests, why not moving all tests to that folder 🤔

Haha I didn't put the E2E in my previous message but that would indeed be for me the only correct usage for a general test folder. The unit tests are tightly coupled to their files, that's why I would put them in the their respective folders.

**Comment by leeuwis** [comment]
When in doubt, I just ask myself WWND*, they also use a separate test folder:
https://github.com/nestjs/nest/tree/0055eb97cf77229427df8fca573797a03dae6979/packages/common/test

However, normally I do think it's nice if something related to a certain component/module is bundled together with that component/module. But with the current structure it became a bit messy in the lib folder itself. If we would add the tests in the lib folder, I think we should create dedicated folder for each decorater/pipe/etc and have the test bundled in there.

_*What Would Nest.JS Do?_

File: test/e2e.test.ts (hunk 1 above)
Comment (leeuwis) [reply test/e2e.test.ts]: Done, but I think we should normalize the descriptions at some points. E.g. always like `it('Should ...')`.

Comment (ggurkal) [reply test/e2e.test.ts]: Agree

**Pull request merged**